wasm_runtime/src/c_api.rs

```diff
@@ -3,254 +3,398 @@
+use crate::execution_ctx::WasmExecutionCtx;
 
-/// Set the root directory for loading Wasm modules.
+
+/// Load a Wasm Module from disk.
+///
+/// All successfully loaded Wasm modules are stored in a `HashMap`.
+/// This implies that:
+///  - The `path` (also used as module's id) must point to an existing file.
+///  - The file pointed by `path` must be a valid .wasm module.
 ///
+/// In case of error, the reason is printed to stderr and returns -1.
+/// Otherwise, it returns 0.
+/// 
 /// Due to String management differences between C and Rust, this function uses `unsafe {}` code.
 /// So `path` must be a valid pointer to a null-terminated C char array. Otherwise, code might panic.
-///
 /// In addition, `path` must contain valid ASCII chars that can be converted into UTF-8 encoding.
-/// Otherwise, the root directory will be an empty string.
 ///
 /// # Examples (C Code)
 ///
 /// ```
-/// wasm_config_set_root("/var/www/wasm");
+/// wasm_module_load("/var/www/wasm/python3_11.wasm");
+/// wasm_module_load("/var/www/wasm/php8.wasm");
 /// ```
 #[no_mangle]
-pub extern "C" fn wasm_config_set_root(path: *const c_char) {
+pub extern "C" fn wasm_module_load(path: *const c_char) -> c_int {
     let path_str = const_c_char_to_str(path);
 
-    let mut config = WASM_RUNTIME_CONFIG.write()
-        .expect("ERROR! Poisoned RwLock WASM_RUNTIME_CONFIG on write()");
-
-    config.path.clear();
-    config.path.push(path_str);
+    match WasmModule::load_from_file(path_str) {
+        Ok(_) => 0,
+        Err(e) => {
+            eprintln!("ERROR! C-API: Couldn't load Wasm module at \"{}\": {}", path_str, e);
+            -1
+        }
+    }
 }
 
-/// Set the Wasm module filename
+
+/// Creates a new Wasm Config given an identifier.
+/// The identifier must be unique.
 ///
+/// In case of error, the reason is printed to stderr and returns -1.
+/// Otherwise, it returns 0.
+/// 
 /// Due to String management differences between C and Rust, this function uses `unsafe {}` code.
-/// So `filename` must be a valid pointer to a null-terminated C char array. Otherwise, code might panic.
-///
-/// In addition, `filename` must contain valid ASCII chars that can be converted into UTF-8 encoding.
-/// Otherwise, the root directory will be an empty string.
+/// So `config_id` must be a valid pointer to a null-terminated C char array. Otherwise, code might panic.
+/// In addition, `config_id` must contain valid ASCII chars that can be converted into UTF-8 encoding.
 ///
 /// # Examples (C Code)
 ///
 /// ```
-/// wasm_config_set_module("hello.wasm");
+/// wasm_config_create("Drupal", "/var/www/php8.wasm");
+/// wasm_config_create("WordPress", "/var/www/php8.wasm");
 /// ```
 #[no_mangle]
-pub extern "C" fn wasm_config_set_module(filename: *const c_char) {
-    let filename_str = const_c_char_to_str(filename);
+pub extern "C" fn wasm_config_create(config_id: *const c_char) -> c_int {
+    let config_id_str = const_c_char_to_str(config_id);
```

Comment:
Documentation is not matching the arguments, docs show two arguments, whereas the function only accepts one.